packages/storage/amplify_storage_s3/example/integration_test/download_file_test.dart

```diff
@@ -0,0 +1,142 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:io';
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:amplify_storage_s3/amplify_storage_s3.dart';
+import 'package:amplify_storage_s3_example/amplifyconfiguration.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+import 'package:path_provider/path_provider.dart';
+
+import 'utils/configure.dart';
+import 'utils/sign_in_new_user.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  group('downloadFile()', () {
+    late String filePath;
+    late List<int> data;
+
+    setUpAll(() async {
+      await configure(amplifyEnvironments['main']!);
+      filePath = 'public/download-file-from-data-${uuid()}';
+      data = 'test data'.codeUnits;
+      await Amplify.Storage.uploadFile(
```

Comment:
todo: use uploadData throughout these tests.

We do not need to test uploadData/uploadFile in these tests as they will have their own tests. uploadData is a simpler, less error prone, API.

Reply:
Apply this change throughout the PR.